### PR TITLE
Consume the diff logic as GitHub job step

### PR DESCRIPTION
**Pull request by pedjak**
### Description of your changes

Add the necessary `action.yaml`, so that the logic can be consumed as a step in an arbitrary GitHub job, simplifying user experience and adoption.

Users need to add something like this to their jobs:

```yaml
jobs:
  example:
  runs-on: ubuntu-latest
  steps:
    - name: Crossplane Dynamic Diff
      uses: upbound/crossplane-diff-action@main
      with:
        # process XRs from the given directory
        dir: examples
        github-token: ${{ secrets.GITHUB_TOKEN }}
```

The action supports an arbitrary target branch.

Other changes:
* removed unneeded action steps
* machine-learning-apps/pr-comment replaced with a custom light script, because the former is not maintained anymore and has a heavy setup part, slowing down the execution.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

File: sort-yaml-content [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env python
```
Comment (haarchri) [inline sort-yaml-content:1]: wonder if we need this after: https://github.com/crossplane/crossplane/pull/5030

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          var issue_number = ${{ github.event.number }};
+          var comment = fs.readFileSync('./diff_content.txt', {encoding:'utf8', flag:'r'});
+          github.rest.issues.createComment({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            issue_number: Number(issue_number),
+            body: comment
+          });
+
+
```
Comment (haarchri) [inline action.yml:110]: can we remove the new lines?

File: render-mrs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    COMPOSITION=$(yq eval '.metadata.annotations."crossplane.io/render-composition-path"' $file)
+    FUNCTION=$(yq eval '.metadata.annotations."crossplane.io/render-function-path"' $file)
+
+    # Skip files where both annotations are null
+    if [[ "$COMPOSITION" == "null" && "$FUNCTION" == "null" ]]; then
+    continue
+    fi
+
+    out=${prefix}${file//\//-}
+    crossplane beta render $file $COMPOSITION $FUNCTION | ${SCRIPT_DIR}/sort-yaml-content > $outdir/$out
```
Comment (haarchri) [inline render-mrs:20]: is this working without an docker/setup-qemu-action@v2 step ? because we starting a docker container for function

Comment (pedjak) [reply render-mrs:20]: docker is already pre-installed on ubuntu GH runners.

File: sort-yaml-content (hunk 1 above)
Comment (pedjak) [reply sort-yaml-content:1]: probably not, but that CLI is not released yet, correct?

**Review by haarchri — APPROVED** [review]
thanks for this enhancement

**Pull request merged**